crates/symbolicator-service/src/services/cacher.rs

```diff
@@ -434,52 +319,131 @@ impl<T: CacheItemRequest> Cacher<T> {
     pub async fn compute_memoized(&self, request: T, cache_key: CacheKey) -> CacheEntry<T::Item> {
         let name = self.config.name();
 
-        // cache_path is None when caching is disabled.
-        if let Some(cache_dir) = self.config.cache_dir() {
-            let versions =
-                std::iter::once(T::VERSIONS.current).chain(T::VERSIONS.fallbacks.iter().copied());
-
-            for version in versions {
-                let item = match self.lookup_local_cache(&request, &cache_key, version) {
-                    Err(CacheError::NotFound) => continue,
-                    Err(err) => return Err(err),
-                    Ok(item) => item,
-                };
-
-                if version != T::VERSIONS.current {
-                    // we have found an outdated cache that we will use right away,
-                    // and we will kick off a recomputation for the `current` cache version
-                    // in a deduplicated background task, which we will not await
-                    tracing::trace!(
-                        "Spawning deduplicated {} computation for path {:?}",
-                        name,
-                        cache_key
-                            .cache_path(cache_dir, T::VERSIONS.current)
-                            .display()
-                    );
-                    metric!(
-                        counter(&format!("caches.{name}.file.fallback")) += 1,
-                        "version" => &version.to_string(),
-                    );
-                    let _not_awaiting_future = self.spawn_computation(request, cache_key, true);
+        // We log quite a few metrics directly:
+        // - caches.X.access
+        // - caches.X.file.hit (emitted inside `lookup_local_cache`)
+        // - caches.X.file.miss
+        // - caches.X.file.write (emitted inside `compute`)
+        // - caches.X.file.fallback
+        // - services.shared_cache.fetch (tagged with `hit`)
+        // Plus some others:
+        // - caches.X.file.size
+        // - services.shared_cache.store
+        // - services.shared_cache.store.bytes
+        // From these we can infer other metrics:
+        // - computations (aka in-memory hit):
+        //   `file.hits + file.miss` which should be the same as `file.write` and `shared_cache.fetch`
+        //   depending on errors and shared cache usage.
+        // - ^ Well actually, eager `computations` should be `file.hits + file.miss - file.fallback`,
+        //   as cache fallback does trigger a background computation.
+        // - in-memory miss: access - computations
+        // FIXME: is it better to use a different metrics key or tags for the cache name?
```

Comment:
NOTE: for datadog it does not matter if we use different keys or tags. I would even prefer to have tags instead, but that might be a good followup.

Reply:
I would agree that tags seem like the natural choice.